open_lm/main.py

```diff
@@ -243,6 +243,15 @@ def main(args):
     # fully initialize distributed device environment
     device = init_distributed_device(args)
 
+    assert (
+        args.global_batch_size % args.world_size == 0
+    ), "Global batch size is not divisible by number of GPUs, and thus cannot be respected."
+
+    args.per_gpu_batch_size = args.global_batch_size // args.world_size
+    if args.val_data is not None:
+        # Make sure that val batch size is set to micro batch size
+        args.per_gpu_val_batch_size = args.global_val_batch_size // args.world_size // args.accum_freq
```

Comment:
Should accum freq be checked ?

Reply:
I changed how `accum_freq` is handled to make it more similar to how it was handled before (in `params.py`). I think this skips the need for checks, unless you have something specific in mind.

Reply:
This looks fine, but just realized, maybe just add a max(..., 0) to ensure we don't accidentally get batch size of 0?